DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -594,6 +604,23 @@ namespace pat {
     void setHcalFraction(float p);                      /// Set the fraction of Ecal and Hcal needed for HF and neutral hadrons
     float hcalFraction() const { return (hcalFraction_/100.); }    /// Fraction of Ecal and Hcal for HF and neutral hadrons
 
+    /// time (wrt nominal zero of the collision)
+    virtual float time(size_t ipv=0)  const { return vertexRef()->t() + dtimeAssociatedPV(); }
+    /// dtime with respect to the PV[ipv]
+    virtual float dtime(size_t ipv=0)  const { return dtimeAssociatedPV() + (*pvRefProd_)[pvRefKey_].t()-(*pvRefProd_)[ipv].t(); }
+    /// dtime with respect to the PV ref
+    virtual float dtimeAssociatedPV()  const {
+        if (packedTime_ == 0) return 0.f;
+        if (packedTimeError_ > 0) return unpackTimeWithError(packedTime_,packedTimeError_);
+        else return unpackTimeNoError(packedTime_);
+    }
+    /// time measurement uncertainty (-1 if not available)
+    virtual float timeError() const { return unpackTimeError(packedTimeError_); }
```

Comment:
wouldn't it be more cost-efficient to fill a transient and make the unpack call only once?